In [1]:
import selenium
import pandas as pd
import time
from selenium import webdriver

# Q1: 
Write a python program to scrape data for “Data Analyst” Job position in “Bangalore” location. You have to scrape the job-title, job-location, company_name, experience_required. You have to scrape first 10 jobs data. This task will be done in following steps:

first get the webpage https://www.naukri.com/
Enter “Data Analyst” in “Skill,Designations,Companies” field and enter “Bangalore” in “enter the location” field.
Then click the search button.
Then scrape the data for the first 10 jobs results you get.
Finally create a dataframe of the scraped data. Note- All of the above steps have to be done in code. No step is to be done manually.

In [2]:
driver1 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url1 = "https://www.naukri.com/"
driver1.get(url1)

In [3]:
search_job1 = driver1.find_element_by_id("qsb-keyword-sugg")
search_job1.send_keys('Data Analyst')

In [4]:
search_location1 = driver1.find_element_by_id("qsb-location-sugg")
search_location1.send_keys("Bangalore")

In [5]:
search_button1 = driver1.find_element_by_xpath("//button[@class='btn']")
search_button1.click()

In [6]:
job_titles1 = driver1.find_elements_by_xpath("//a[@class = 'title fw500 ellipsis']")
titles1 = []
for i in job_titles1:
    titles1.append(i.text)
#titles1

company_name1 = driver1.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company1 = []
for i in company_name1:
    company1.append(i.text)
#company1

experience1 = driver1.find_elements_by_xpath("//li[@class = 'fleft grey-text br2 placeHolderLi experience']//span")
exp1 = []
for i in experience1:
    exp1.append(i.text)
#exp1

job_location1 = driver1.find_elements_by_xpath("//li[@class = 'fleft grey-text br2 placeHolderLi location']//span")
location1 = []
for i in job_location1:
    location1.append(i.text)
#location1

In [7]:
jobs1 = pd.DataFrame()
jobs1['Job Title'] = titles1
jobs1['Company Name'] = company1
jobs1['Experience Required'] = exp1
jobs1['Location'] = location1
jobs1[:10]

Job Title  \
0    Data Scientist / Data Analyst -Business Analyst   
1  Hiring Data Analysts For E commerce Platform |...   
2                                       Data Analyst   
3  Hiring For Data Analyst/ MIS Reporting Analyst...   
4  DA - Urgent Opening For Data Analyst BFSI Doma...   
5                     Data Analyst - Informatica MDM   
6  Assistant Vice President - MIS & Reporting ( B...   
7                                       Data Analyst   
8                                       Data Analyst   
9                                       Data Analyst   

                                        Company Name Experience Required  \
0                 Inflexion Analytix Private Limited             0-3 Yrs   
1                   Allegis Services India Pvt. Ltd.             0-5 Yrs   
2                                  Applied Materials            7-10 Yrs   
3   PHARMACEUTICAL RESEARCH ASSOCIATES INDIA Pvt Ltd             2-4 Yrs   
4                     Tata Consultancy Services Ltd.             4-9 Yrs   
5                Shell India Markets Private Limited             6-9 Yrs   
6  INTERTRUSTVITEOS CORPORATE AND FUND SERVICES P...           12-18 Yrs   
7                           Myntra Designs Pvt. Ltd.             3-6 Yrs   
8                           Myntra Designs Pvt. Ltd.             3-6 Yrs   
9                           Myntra Designs Pvt. Ltd.             4-9 Yrs   

                                            Location  
0  Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/...  
1                                Bangalore/Bengaluru  
2                                Bangalore/Bengaluru  
3                                Bangalore/Bengaluru  
4  Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...  
5                                Bangalore/Bengaluru  
6                        Mumbai, Bangalore/Bengaluru  
7                                Bangalore/Bengaluru  
8                                Bangalore/Bengaluru  
9                                Bangalore/Bengaluru

# Q2: 
Write a python program to scrape data for “Data Scientist” Job position in
“Bangalore” location. You have to scrape the job-title, job-location,
company_name, full job-description. You have to scrape first 10 jobs data.
This task will be done in following steps:
1. first get the webpage https://www.naukri.com/
2. Enter “Data Scientist” in “Skill,Designations,Companies” field and enter “Bangalore” in “enter the location” field.
3. Then click the search button.
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.

In [8]:
driver2 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url2 = "https://www.naukri.com/"
driver2.get(url2)

In [9]:
search_job2 = driver2.find_element_by_id("qsb-keyword-sugg")
search_job2.send_keys('Data Scientist')
search_location2 = driver2.find_element_by_id("qsb-location-sugg")
search_location2.send_keys("Bangalore")
search_button2 = driver2.find_element_by_xpath("//button[@class='btn']")
search_button2.click()

In [10]:
# Scrapping for Job Title
job_titles2 = driver2.find_elements_by_xpath("//a[@class = 'title fw500 ellipsis']")
titles2 = []
for i in job_titles2:
    titles2.append(i.text)
#titles2

# Scrapping for Company Name
company_name2 = driver2.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company2 = []
for i in company_name2:
    company2.append(i.text)
#company2

# Scrapping for Job location
job_location2 = driver2.find_elements_by_xpath("//li[@class = 'fleft grey-text br2 placeHolderLi location']//span")
location2 = []
for i in job_location2:
    location2.append(i.text)
#location2

In [11]:
# Scrapping for total job description

# Finding the urls for each job
job_urls2 = []
job_url2 = driver2.find_elements_by_xpath("//a[@class = 'title fw500 ellipsis']")
for i in job_url2:
    job_urls2.append(i.get_attribute("href"))
#job_urls2
job_urls2 = job_urls2[:10]

desc2 = []
from selenium.common.exceptions import NoSuchElementException # To handle no such element exception
for i in job_urls2:
    driver2.get(i)
    time.sleep(3)
    try:
        job_description2 = driver2.find_element_by_xpath("//section[@class = 'job-desc']")
        desc2.append(job_description2.text.replace('\n',''))
    except NoSuchElementException:
        desc2.append('-')
#desc2

In [12]:
jobs2 = pd.DataFrame()
jobs2['Job Title'] = titles2
jobs2['Company Name'] = company2
jobs2['Location'] = location2
jobs2 = jobs2.iloc[:10,:]
jobs2['Total Description'] = desc2
jobs2

Job Title  \
0    Data Scientist / Data Analyst -Business Analyst   
1                    Senior Data Scientist, Modeling   
2                        Data Scientist - IBM Garage   
3                                     Data Scientist   
4                Senior Data Scientist - Credit risk   
5                          Big Data - Data Scientist   
6                                     Data Scientist   
7                                Lead Data Scientist   
8  Hiring For DATA Scientist - ON Contract Basis ...   
9              Senior Data Scientist - Chatbot & NLP   

                                        Company Name  \
0                 Inflexion Analytix Private Limited   
1                                            Nielsen   
2                             IBM India Pvt. Limited   
3                             IBM India Pvt. Limited   
4                                 Scienaptic Systems   
5                          Xoriant Solutions Pvt Ltd   
6                             IBM India Pvt. Limited   
7                     Intel Technology India Pvt Ltd   
8  GlobalEdx Learning and Technology Solution Pvt...   
9                                         Gojek Tech   

                                            Location  \
0  Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/...   
1  Kolkata, Gurgaon/Gurugram, Bangalore/Bengaluru...   
2  Noida, Hyderabad/Secunderabad, Bangalore/Benga...   
3  Noida, Hyderabad/Secunderabad, Bangalore/Benga...   
4                                Bangalore/Bengaluru   
5  Kochi/Cochin, Indore, Hyderabad/Secunderabad, ...   
6                                Bangalore/Bengaluru   
7                                Bangalore/Bengaluru   
8  Hyderabad/Secunderabad, Bangalore/Bengaluru, M...   
9                                Bangalore/Bengaluru   

                                   Total Description  
0  Job descriptionJob Role : Data Scientist/Data ...  
1  Job descriptionWe wont say we can predict the ...  
2                                                  -  
3                                                  -  
4  Job descriptionResponsibilities and duties Foc...  
5                                                  -  
6                                                  -  
7  Job descriptionWe are seeking an outstanding L...  
8  Job descriptionDear Aspirant,Greetings from Gl...  
9  Job descriptionWhat You Will DoWork with Data ...

# Q3: 
In this question you have to scrape data using the filters available on the
webpage as shown below:
You have to use the location and salary filter.
You have to scrape data for “Data Scientist” designation for first 10 job results.
You have to scrape the job-title, job-location, company_name,
experience_required.
The location filter to be used is “Delhi/NCR”
The salary filter to be used is “3-6” lakhs
The task will be done as shown in the below steps:
1. first get the webpage https://www.naukri.com/
2. Enter “Data Scientist” in “Skill,Designations,Companies” field .
3. Then click the search button.
4. Then apply the location filter and salary filter by checking the respective boxes
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.

Note- All of the above steps have to be done in code. No step is to be done
manually.

In [18]:
driver3 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url3 = "https://www.naukri.com/"
driver3.get(url3)

In [19]:
# Entering title and location on webpage.
search_job3 = driver3.find_element_by_id("qsb-keyword-sugg")
search_job3.send_keys('Data Scientist')
search_location3 = driver3.find_element_by_id("qsb-location-sugg")
search_location3.send_keys("Delhi/NCR")
search_button3 = driver3.find_element_by_xpath("//button[@class='btn']")
search_button3.click()

In [20]:
# Enterning the filters required
location_filter3 = driver3.find_element_by_xpath("//label[@for = 'chk-Delhi / NCR-cityTypeGid-']//i")
location_filter3.click()
time.sleep(5)
salary_filter3 = driver3.find_element_by_xpath("//label[@for = 'chk-3-6 Lakhs-ctcFilter-']//i")
salary_filter3.click()

In [21]:
# Scrapping for job titles
job_titles3 = driver3.find_elements_by_xpath("//a[@class = 'title fw500 ellipsis']")
titles3 = []
for i in job_titles3:
    titles3.append(i.text)
#titles3

# Scrapping for company name
company_name3 = driver3.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company3 = []
for i in company_name3:
    company3.append(i.text)
#company3

# Scrapping for experience required
experience3 = driver3.find_elements_by_xpath("//li[@class = 'fleft grey-text br2 placeHolderLi experience']//span")
exp3 = []
for i in experience3:
    exp3.append(i.text)
#exp3

# Scrapping for job location
job_location3 = driver3.find_elements_by_xpath("//li[@class = 'fleft grey-text br2 placeHolderLi location']//span")
location3 = []
for i in job_location3:
    location3.append(i.text)
#location3

In [22]:
jobs3 = pd.DataFrame()
jobs3['Job Title'] = titles3
jobs3['Company Name'] = company3
jobs3['Experience Required'] = exp3
jobs3['Job Location'] = location3
jobs3[:10]

Job Title  \
0    Data Scientist / Data Analyst -Business Analyst   
1  Data Scientist - High growth VC backed Influen...   
2           Excellent opportunity For Data Scientist   
3                                     Data Scientist   
4           DATA Scientist – Gurgaon (Exp 3-6 years)   
5           DATA Scientist – Gurgaon (Exp 3-6 years)   
6                             Data Scientist - Noida   
7                     Data Scientist - Noida/ B'lore   
8                                     Data Scientist   
9                                     Data Scientist   

                                        Company Name Experience Required  \
0                 Inflexion Analytix Private Limited             0-3 Yrs   
1                    Ravgins International Pvt. Ltd.             3-5 Yrs   
2              NEC CORPORATION INDIA PRIVATE LIMITED             3-7 Yrs   
3                                           Mobikwik             3-5 Yrs   
4  CRESCENDO GLOBAL LEADERSHIP HIRING INDIA PRIVA...             3-6 Yrs   
5  CRESCENDO GLOBAL LEADERSHIP HIRING INDIA PRIVA...             3-6 Yrs   
6     Optum Global Solutions (India) Private Limited             3-5 Yrs   
7              NEC CORPORATION INDIA PRIVATE LIMITED             3-8 Yrs   
8           Cloudstrats Technologies Private Limited             5-8 Yrs   
9                             IBM India Pvt. Limited             4-9 Yrs   

                                        Job Location  
0  Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/...  
1  Bangalore/Bengaluru, Delhi / NCR, Mumbai (All ...  
2                         Noida, Bangalore/Bengaluru  
3           New Delhi, Gurgaon/Gurugram, Delhi / NCR  
4                      Gurgaon/Gurugram, Delhi / NCR  
5                      Gurgaon/Gurugram, Delhi / NCR  
6                                              Noida  
7                         Noida, Bangalore/Bengaluru  
8  Bangalore/Bengaluru, Delhi / NCR, Mumbai (All ...  
9  Noida, Hyderabad/Secunderabad, Bangalore/Benga...

# Q4: 
Write a python program to scrape data for first 10 job results for Data scientist
Designation in Noida location. You have to scrape company_name, No. of days
ago when job was posted, Rating of the company.
This task will be done in following steps:
1. first get the webpage https://www.glassdoor.co.in/index.htm
2. Enter “Data Scientist” in “Job Title,Keyword,Company” field and enter “Noida”
in “location” field.
3. Then click the search button. You will land up in the below page:
4. Then scrape the data for the first 10 jobs results you get in the above shown page.
5. Finally create a dataframe of the scraped data.

Note- All of the above steps have to be done in code. No step is to be done
manually.

In [23]:
# Opening the chrome driver pop-up 
driver4 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url4 = "https://www.glassdoor.co.in/Job/Home/recentActivity.htm"
driver4.get(url4)
print("\033[1mPLEASE SIGN IN FOR THE WEBSITE IN THE OPEN CHROME DRIVER POP_UP !!!\033[0m")

PLEASE SIGN IN FOR THE WEBSITE IN THE OPEN CHROME DRIVER POP_UP !!!


In [24]:
# putting in the search values
# searching job title
search_job4 = driver4.find_element_by_id("sc.keyword")
search_job4.send_keys('Data Scientist')
time.sleep(1)

# searching location
search_location4 = driver4.find_element_by_id("sc.location")
search_location4.clear()
time.sleep(1)
search_location4.send_keys("Noida")

#clicking search button
search_button4 = driver4.find_element_by_xpath("//button[@class='gd-ui-button ml-std col-auto SearchStyles__newSearchButton css-iixdfr']")
search_button4.click()

In [25]:
# Scrapping for Job Titles 
job_titles4 = driver4.find_elements_by_xpath("//a[@class = 'jobLink css-1rd3saf eigr9kq2']//span")
titles4 = []
for i in job_titles4:
    titles4.append(i.text)
#titles4

# Scrapping for company name
company_name4 = driver4.find_elements_by_xpath("//a[@class=' css-l2wjgv e1n63ojh0 jobLink']//span")
company4 = []
for i in company_name4:
    company4.append(i.text)
#company4

# Scrapping for days of job posting
no_of_days4 = driver4.find_elements_by_xpath("//div[@data-test = 'job-age']")
days4 = []
for i in no_of_days4:
    days4.append(i.text)
#days4

# Scrapping for company rating
company_rating4 = driver4.find_elements_by_xpath("//div[@class = 'd-flex flex-column css-x75kgh e1rrn5ka3']")
rating4 = []
for i in company_rating4:
    rating4.append(i.text)
#rating4

In [26]:
jobs4 = pd.DataFrame()
jobs4['Job Title'] = titles4
jobs4['Company Name'] = company4
jobs4['Company Rating'] = rating4
jobs4['No. of Days Old'] = days4
jobs4[:10]

Job Title  \
0                         Lead Analyst- Data Science   
1                                     Data Scientist   
2                                     Data Scientist   
3                                     Data Scientist   
4                        Lead- Python (Data Science)   
5                Data Scientist (BE/ B-tech 4-6 yrs)   
6                                     Data Scientist   
7  Data Analytics part time job/internship at Ban...   
8                        Data Scientist - IBM Garage   
9                                 RAU-Data Scientist   

                                        Company Name Company Rating  \
0                                    Infosys Limited            3.8   
1                             Applied Materials Inc.            4.0   
2                                                GSK            3.8   
3                                                IBM            3.9   
4                                    Infosys Limited            3.8   
5                                         CommerceIQ            3.8   
6  Mercedes-Benz Research and Development India P...            4.0   
7                                           CueLearn            3.9   
8                                                IBM            3.9   
9                                          HDFC Bank            3.6   

  No. of Days Old  
0            30d+  
1            30d+  
2             24h  
3              3d  
4            30d+  
5                  
6              8d  
7             10d  
8             17d  
9             25d

# Q5: 
Write a python program to scrape the salary data for Data Scientist designation
in Noida location.
You have to scrape Company name, Number of salaries, Average salary, Min
salary, Max Salary.
The above task will be, done as shown in the below steps:
1. first get the webpage https://www.glassdoor.co.in/Salaries/index.htm
2. Enter “Data Scientist” in Job title field and “Noida” in location field.
3. Click the search button.
4. After that you will land on the below page. You have to scrape whole data from this webpage
5. Scrape data for first 10 companies. Scrape the min salary, max salary, company name, Average salary and rating of the company.
6. Store the data in a dataframe.

Note that all of the above steps have to be done by coding only and not manually.

In [27]:
# opening the webpage on chrome driver.
driver5 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url5 = "https://www.glassdoor.co.in/Salaries/index.htm"
driver5.get(url5)

In [28]:
# putting in the search values
# searching job title
search_job5 = driver5.find_element_by_id("KeywordSearch")
search_job5.send_keys('Data Scientist')
time.sleep(1)

# searching location
search_location5 = driver5.find_element_by_id("LocationSearch")
search_location5.clear()
time.sleep(1)
search_location5.send_keys("Noida")

#clicking search button
search_button5 = driver5.find_element_by_xpath("//button[@class='gd-btn-mkt']")
search_button5.click()

In [29]:
# Scrapping for company name
company_name5 = driver5.find_elements_by_xpath("//p[@class = 'm-0 ']")
company5 = []
for i in company_name5:
    company5.append(i.text)
#company5

# Scrapping for maximum and minimum salary
salary5 = driver5.find_elements_by_xpath("//div[@class = 'common__RangeBarStyle__values d-flex justify-content-between ']")
salaries5 = []
for i in salary5:
    salaries5.append(i.text)
#salaries5 - list with both maximum and minimum salaries.

'''indexing to obtain max and min salaries'''
min_salary5 = []
for i in salaries5:
    min_salary5.append(i.split('\n')[0])
#min_salary5

max_salary5 = []
for i in salaries5:
    max_salary5.append(i.split('\n')[1])
#max_salary5

# Scrapping for average salary.
avg_sal5 = driver5.find_elements_by_xpath("//div[@class = 'col-2 d-none d-md-flex flex-row justify-content-end']")
avgsal5 = []
for i in avg_sal5:
    avgsal5.append(i.text.replace('\n', ''))
#avgsal5

In [31]:
jobs5 = pd.DataFrame()
jobs5['Company Name'] = company5
jobs5['Average Salary'] = avgsal5
jobs5['Minimum Salary'] = min_salary5
jobs5['Maximum Salary'] = max_salary5
jobs5[:10]

Company Name  Average Salary Minimum Salary Maximum Salary
0  Tata Consultancy Services   ₹ 6,11,228/yr          ₹343K        ₹1,095K
1                  Accenture  ₹ 11,46,533/yr          ₹577K        ₹2,213K
2                        IBM   ₹ 8,97,795/yr          ₹586K        ₹2,730K
3         Ericsson-Worldwide   ₹ 7,38,057/yr          ₹355K        ₹1,613K
4                  Delhivery  ₹ 12,39,781/yr          ₹450K       ₹11,622K
5         UnitedHealth Group  ₹ 13,36,142/yr        ₹1,069K        ₹1,520K
6         Valiance Solutions   ₹ 8,15,192/yr          ₹502K        ₹1,465K
7              ZS Associates  ₹ 11,35,221/yr          ₹202K        ₹1,809K
8                EXL Service  ₹ 11,44,243/yr          ₹575K        ₹1,520K
9     Optum Global Solutions  ₹ 14,13,288/yr        ₹1,014K        ₹2,149K

# Q6 : 
Scrape data of first 100 sunglasses listings on flipkart.com. You have to
scrape four attributes:
1. Brand
2. Product Description
3. Price
4. Discount %

To scrape the data you have to go through following steps:
1. Go to flipkart webpage by url https://www.flipkart.com/
2. Enter “sunglasses” in the search field where “search for products, brands and more” is written and click the search icon
3. after that you will reach to a webpage having a lot of sunglasses. From this page you can scrap the required data as usual.
4. after scraping data from the first page, go to the “Next” Button at the bottom of the page , then click on it
5. Now scrape data from this page as usual
6. repeat this until you get data for 100 sunglasses.

Note that all of the above steps have to be done by coding only and not manually.

In [32]:
# opening the webpage on chrome driver
driver6 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url6 = "https://www.flipkart.com/"
driver6.get(url6)

In [33]:
# login cross button
cross_button6 = driver6.find_element_by_xpath("//button[@class = '_2KpZ6l _2doB4z']")
cross_button6.click()

In [34]:
# putting in the search values
# searching sunglasses
search_item6 = driver6.find_element_by_xpath("//input[@class = '_3704LK']")
search_item6.send_keys('Sunglasses')
time.sleep(1)

#clicking search button
search_button6 = driver6.find_element_by_xpath("//button[@class='L0Z3Pu']")
search_button6.click()

In [35]:
# Scrapping the brand name from first page
brand_name6 = driver6.find_elements_by_xpath("//div[@class = '_2WkVRV']")
brand6 = []
for i in brand_name6:
    brand6.append(i.text)
#brand6

# Scrapping product description from first page
product_desc6 = driver6.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
description6 = []
for i in product_desc6:
    description6.append(i.text)
#description6

# Scrapping price from first page
price6 = driver6.find_elements_by_xpath("//div[@class = '_30jeq3']")
mrp6 = []
for i in price6:
    mrp6.append(i.text)
#mrp6

# SCrapping discount from first page
discount6 = driver6.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")
disc6 = []
for i in discount6:
    disc6.append(i.text)
#disc6

print(f"Total number of products from first page: {len(brand6)}")

Total number of products from first page: 40


We know that we have to scrap the total of 100 product for this problem, so we move onto next page and see how many products we have got.

In [36]:
if len(brand6) < 100:
    next_button6 = driver6.find_element_by_xpath("//a[@class = '_1LKTO3']//span")
    next_button6.click()
    time.sleep(2)
    brand_name6 = driver6.find_elements_by_xpath("//div[@class = '_2WkVRV']")
    product_desc6 = driver6.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
    price6 = driver6.find_elements_by_xpath("//div[@class = '_30jeq3']")
    discount6 = driver6.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")   
    for i in brand_name6:
        brand6.append(i.text)
    for i in product_desc6:
        description6.append(i.text)
    for i in price6:
        mrp6.append(i.text)
    for i in discount6:
        disc6.append(i.text)
    print(f"List appended. Total Products: {len(brand6)}")

List appended. Total Products: 80


We go again and again until the total number of products are achieved.

In [37]:
if len(brand6) < 100:
    next_button6 = driver6.find_element_by_xpath("//a[@class = '_1LKTO3'][2]")
    next_button6.click()
    time.sleep(2)
    brand_name6 = driver6.find_elements_by_xpath("//div[@class = '_2WkVRV']")
    product_desc6 = driver6.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
    price6 = driver6.find_elements_by_xpath("//div[@class = '_30jeq3']")
    discount6 = driver6.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")   
    for i in brand_name6:
        brand6.append(i.text)
    for i in product_desc6:
        description6.append(i.text)
    for i in price6:
        mrp6.append(i.text)
    for i in discount6:
        disc6.append(i.text)
    print(f"List appended. Total Products: {len(brand6)}")
else:
    print(f"Total products: {len(brand6)}")

List appended. Total Products: 120


We see that we have got total of 120 products, we can now save the complete data into the dataframe.

In [38]:
products6 = pd.DataFrame()
products6['Product'] = brand6[:100]
products6['Description'] = description6[:100]
products6['Discount'] = disc6[:100]
products6['Price after discount'] = mrp6[:100]
products6

Product                                        Description  \
0           AISLIN  UV Protection, Gradient Round, Shield Sunglass...   
1        ROYAL SON         UV Protection Retro Square Sunglasses (49)   
2         Fastrack   UV Protection Rectangular Sunglasses (Free Size)   
3         Fastrack      UV Protection Wayfarer Sunglasses (Free Size)   
4   ROZZETTA CRAFT  UV Protection Retro Square Sunglasses (Free Size)   
..             ...                                                ...   
95          FOSSIL                 Others Rectangular Sunglasses (54)   
96       Elligator                UV Protection Round Sunglasses (54)   
97       ROYAL SON                   Mirrored Aviator Sunglasses (58)   
98       ROYAL SON         UV Protection Retro Square Sunglasses (58)   
99           NuVew  UV Protection, Gradient, Night Vision, Mirrore...   

   Discount Price after discount  
0   77% off                 ₹395  
1   66% off                 ₹664  
2   21% off                 ₹630  
3   15% off                 ₹758  
4   77% off                 ₹499  
..      ...                  ...  
95  55% off               ₹2,625  
96  88% off                 ₹279  
97  73% off                 ₹399  
98  66% off                 ₹499  
99  69% off                 ₹377  

[100 rows x 4 columns]

# Q7: 
Scrape 100 reviews data from flipkart.com for iphone11 phone. You have to
go the link: https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/p/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace.
When you will open the above link you will reach to the below shown webpage.
As shown in the above page you have to scrape the tick marked attributes.
These are
1. Rating
2. Review_summary
3. Full review
You have to scrape this data for first 100 reviews.

In [10]:
driver7 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url7 = "https://www.flipkart.com/apple-iphone-11-black-64-gb-includes-earpods-power-adapter/p/itm0f37c2240b217?pid=MOBFKCTSVZAXUHGR&lid=LSTMOBFKCTSVZAXUHGREPBFGI&marketplace"
driver7.get(url7)

In [11]:
# Getting all the revies using all reviews button
all_rev_button = driver7.find_element_by_xpath("//div[@class = '_3UAT2v _16PBlm']")
time.sleep(3)
all_rev_button.click()

In [12]:
# Scrapping for the ratings from the first webpage.
rating7 = []
product_rating7 = driver7.find_elements_by_xpath("//div[@class = 'col _2wzgFH K0kLPL']//div[@class= 'row'][1]//div[1]")
for i in product_rating7:
    rating7.append(i.text)
#rating7 - we get a total of 60 values which consist invalid values along the ratings. We will slice the list at the end.

# Scrappiing for the review summary
product_review7 = driver7.find_elements_by_xpath("//p[@class = '_2-N8zT']")
review7 = []
for i in product_review7:
    review7.append(i.text)
#review7

# Scrapping for the total full review
product_full_review7 = driver7.find_elements_by_xpath("//div[@class = 't-ZTKy']")
full_review7 = []
for i in product_full_review7:
    full_review7.append(i.text.replace('\n',' ').replace('❤️','').replace('😃', ''))
#full_review7

In [13]:
# Going to the second web page.
if len(review7) < 100:
    next_button7 = driver7.find_element_by_xpath("//a[@class = '_1LKTO3']//span")
    next_button7.click() # clicking the next button
    time.sleep(3)
    product_rating7 = driver7.find_elements_by_xpath("//div[@class = 'col _2wzgFH K0kLPL']//div[@class= 'row'][1]//div[1]")
    product_review7 = driver7.find_elements_by_xpath("//p[@class = '_2-N8zT']")
    product_full_review7 = driver7.find_elements_by_xpath("//div[@class = 't-ZTKy']") 
    for i in product_rating7:
        rating7.append(i.text) # appending the list again
    for i in product_review7:
        review7.append(i.text) # appending the list again
    for i in product_full_review7:
        full_review7.append(i.text.replace('\n',' ').replace('❤️','').replace('😃', '')) # appending the list again

    print(f"List appended. Lists length: {len(review7)}")

List appended. Lists length: 20


In [14]:
'''Assigning a while loop to go to next page and scrap data from that page until we have a total of 1000 reviews of the product'''

while len(review7) <= 100: # while loop to repeat clicking to go to next page until 100 reviews scrapped
    if len(review7) < 100:
        next_button7 = driver7.find_element_by_xpath("//a[@class = '_1LKTO3'][2]")
        next_button7.click()
        time.sleep(3)
        product_rating7 = driver7.find_elements_by_xpath("//div[@class = 'col _2wzgFH K0kLPL']//div[@class= 'row'][1]//div[1]")
        product_review7 = driver7.find_elements_by_xpath("//p[@class = '_2-N8zT']")
        product_full_review7 = driver7.find_elements_by_xpath("//div[@class = 't-ZTKy']")   
        for i in product_rating7:
            rating7.append(i.text)
        for i in product_review7:
            review7.append(i.text)
        for i in product_full_review7:
            full_review7.append(i.text.replace('\n',' ').replace('❤️','').replace('😃', ''))
        time.sleep(1)
        print(f"List appended. Lists length: {len(review7)}")
        continue
    else:
        print(f"Total reviews: {len(review7)}")
        break

List appended. Lists length: 30
List appended. Lists length: 40
List appended. Lists length: 50
List appended. Lists length: 60
List appended. Lists length: 70
List appended. Lists length: 80
List appended. Lists length: 90
List appended. Lists length: 100
Total reviews: 100


We know that for the list of ratings, we have 60 values from each page. So, we need every 6th element from that list. Using the sliing on the rating7 list to get the desired output.

In [15]:
rating7 = rating7[::6]
print(rating7)

['5', '5', '5', '5', '4', '5', '5', '5', '5', '5', '5', '5', '5', '5', '4', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '4', '1', '5', '5', '5', '5', '5', '5', '5', '5', '4', '4', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '4', '5', '5', '5', '5', '5', '5', '5', '5', '5', '4', '4', '5', '5', '5', '4', '5', '5', '3', '5', '5', '5', '5', '5', '5', '5', '5', '4', '4', '5', '1', '3', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5', '5']


We see that we have got the desired rating, now making a dataframe of all the data.

In [16]:
phone_rating7 = pd.DataFrame()
phone_rating7['Ratings'] = rating7
phone_rating7['Review Summary'] = review7
phone_rating7['Full Review'] = full_review7
phone_rating7

Ratings       Review Summary  \
0        5            Brilliant   
1        5     Perfect product!   
2        5        Great product   
3        5    Worth every penny   
4        4          Good choice   
..     ...                  ...   
95       5  Best in the market!   
96       5             Terrific   
97       5    Terrific purchase   
98       5            Wonderful   
99       5            Fabulous!   

                                          Full Review  
0   The Best Phone for the Money  The iPhone 11 of...  
1   Amazing phone with great cameras and better ba...  
2   Amazing Powerful and Durable Gadget.  I’m am v...  
3   Previously I was using one plus 3t it was a gr...  
4   So far it’s been an AMAZING experience coming ...  
..                                                ...  
95  The best all rounder iphone. Flipkart is doing...  
96  I upgraded (downgraded?) from my iPhone X sinc...  
97  Well , as we all know if its not an Iphone , i...  
98  Nice value for money good and best price I pho...  
99  Just an awesome phone...upgraded from 6s to 11...  

[100 rows x 3 columns]

# Q8: 
Scrape data for first 100 sneakers you find when you visit flipkart.com and
search for “sneakers” in the search field.
You have to scrape 4 attributes of each sneaker :
1. Brand
2. Product Description
3. Price
4. discount %

Also note that all the steps required during scraping should be done through code
only and not manually.

In [17]:
driver8 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url8 = "https://www.flipkart.com/"
driver8.get(url8)

In [18]:
cross_button8 = driver8.find_element_by_xpath("//button[@class = '_2KpZ6l _2doB4z']")
cross_button8.click()

In [19]:
# putting in the search values
# searching sneakers
search_item8 = driver8.find_element_by_xpath("//input[@class = '_3704LK']")
search_item8.send_keys('sneakers')
time.sleep(1)

#clicking search button
search_button8 = driver8.find_element_by_xpath("//button[@class='L0Z3Pu']")
search_button8.click()

In [20]:
# Scrapping the brand name from first page
brand_name8 = driver8.find_elements_by_xpath("//div[@class = '_2WkVRV']")
brand8 = []
for i in brand_name8:
    brand8.append(i.text)
#brand8

# Scrapping product description from first page
product_desc8 = driver8.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
description8 = []
for i in product_desc8:
    description8.append(i.text)
#description8

# Scrapping price from first page
price8 = driver8.find_elements_by_xpath("//div[@class = '_30jeq3']")
mrp8 = []
for i in price8:
    mrp8.append(i.text)
#mrp8

# SCrapping discount from first page
discount8 = driver8.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")
disc8 = []
for i in discount8:
    disc8.append(i.text)
#disc8

print(f"Total number of products from first page: {len(brand8)}")

Total number of products from first page: 40


In [21]:
if len(brand8) < 100:
    next_button8 = driver8.find_element_by_xpath("//a[@class = '_1LKTO3']//span")
    next_button8.click()
    time.sleep(2)
    brand_name8 = driver8.find_elements_by_xpath("//div[@class = '_2WkVRV']")
    product_desc8 = driver8.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
    price8 = driver8.find_elements_by_xpath("//div[@class = '_30jeq3']")
    discount8 = driver8.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")   
    for i in brand_name8:
        brand8.append(i.text)
    for i in product_desc8:
        description8.append(i.text)
    for i in price8:
        mrp8.append(i.text)
    for i in discount8:
        disc8.append(i.text)
    print(f"List appended. Total Products: {len(brand8)}")

List appended. Total Products: 80


In [22]:
if len(brand8) < 100:
    next_button8 = driver8.find_element_by_xpath("//a[@class = '_1LKTO3'][2]")
    next_button8.click()
    time.sleep(2)
    brand_name8 = driver8.find_elements_by_xpath("//div[@class = '_2WkVRV']")
    product_desc8 = driver8.find_elements_by_xpath("//div[@class = '_2B099V']//a[1]")
    price8 = driver8.find_elements_by_xpath("//div[@class = '_30jeq3']")
    discount8 = driver8.find_elements_by_xpath("//div[@class = '_3Ay6Sb']")   
    for i in brand_name8:
        brand8.append(i.text)
    for i in product_desc8:
        description8.append(i.text)
    for i in price8:
        mrp8.append(i.text)
    for i in discount8:
        disc8.append(i.text)
    print(f"List appended. Total Products: {len(brand8)}")
else:
    print(f"Total products: {len(brand8)}")

List appended. Total Products: 120


In [24]:
products8 = pd.DataFrame()
products8['Product'] = brand8[:100]
products8['Description'] = description8[:100]
products8['Discount'] = disc8[:100]
products8['Price after discount'] = mrp8[:100]
products8

Product                                        Description  \
0   luxury fashion              Luxury sneaker shoes Sneakers For Men   
1        India hub       Canvas casual sneaker shoes Sneakers For Men   
2     Robbie jones     Casual Sneakers Shoes For Men Sneakers For Men   
3          Numenzo                                   Sneakers For Men   
4           ORICUM  Combo pack of 2 casual sneaker shoes for men S...   
..             ...                                                ...   
95           Xtoon                      Casual shoes Sneakers For Men   
96    BLACK BOTTOM  Fashion Outdoor Canvas Casual Light Weight Par...   
97        HOTSTYLE                          Sneakers Sneakers For Men   
98        Bersache  Combo Pack of 5 Top Rated, Best Rates Training...   
99          BRUTON      Combo Pack of 2 Casual Shoes Sneakers For Men   

   Discount Price after discount  
0   80% off                 ₹599  
1   71% off                 ₹569  
2   62% off                 ₹379  
3   60% off                 ₹398  
4   62% off                 ₹377  
..      ...                  ...  
95  64% off                 ₹359  
96  55% off                 ₹449  
97  40% off                 ₹299  
98  71% off                 ₹719  
99  82% off                 ₹425  

[100 rows x 4 columns]

# Q9: 
Go to the link - https://www.myntra.com/shoes
Set Price filter to “Rs. 6649 to Rs. 13099” , Color filter to “Black”

And then scrape First 100 shoes data you get. The data should include “Brand” of
the shoes , Short Shoe description, price of the shoe.

Please note that applying the filter and scraping the data , everything should be
done through code only and there should not be any manual step.

In [25]:
driver9 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url = "https://www.myntra.com/shoes"
driver9.get(url)

In [26]:
# applying colour filter
colour_filter9 = driver9.find_element_by_xpath("//span[@data-colorhex = 'black']")
colour_filter9.click()

In [27]:
# applying price filter
price_filter9 = driver9.find_element_by_xpath("//*[@id='mountRoot']/div/div[1]/main/div[3]/div[1]/section/div/div[5]/ul/li[2]/label/div")
price_filter9.click()

In [28]:
# Scrapping the brand of shoe
shoe_brand9 = driver9.find_elements_by_xpath("//div[@class = 'product-productMetaInfo']//h3[@class = 'product-brand']")
brand9 = []
for i in shoe_brand9:
    brand9.append(i.text)
#brand9

# scrapping for product description
product_desc9 = driver9.find_elements_by_xpath("//div[@class = 'product-productMetaInfo']//h4[@class = 'product-product']")
desc9 = []
for i in product_desc9:
    desc9.append(i.text)
#desc9

# Scrapping the price of the shoes
shoe_price9 = driver9.find_elements_by_xpath("//div[@class = 'product-price']")
price9 = []
for i in shoe_price9:
    price9.append(i.text[:9].replace('Rs. ', '').replace('R', ''))
#price9

print(f"List appended. Total Products: {len(brand9)}")

List appended. Total Products: 50


We need to scrap the data for 100 shoes so we move on to next page.

In [29]:
if len(brand9) < 100: # if number of shoes is less than 100
    next_button9 = driver9.find_element_by_xpath("//a[@rel = 'next']")
    next_button9.click() # clicking the button for next page
    
    time.sleep(2)
    
    shoe_brand9 = driver9.find_elements_by_xpath("//div[@class = 'product-productMetaInfo']//h3[@class = 'product-brand']")
    product_desc9 = driver9.find_elements_by_xpath("//div[@class = 'product-productMetaInfo']//h4[@class = 'product-product']")
    shoe_price9 = driver9.find_elements_by_xpath("//div[@class = 'product-price']")   
    
    for i in shoe_brand9:
        brand9.append(i.text)
    
    for i in product_desc9:
        desc9.append(i.text)
    
    for i in shoe_price9:
        price9.append(i.text[:9].replace('Rs. ', '').replace('R', ''))
    
    print(f"List appended. Total Products: {len(brand9)}")

else: # if number of shoes is more than 100
    print(f"Total products: {len(brand9)}")

List appended. Total Products: 100


In [30]:
products9 = pd.DataFrame()
products9['Show Brand'] = brand9
products9['Description'] = desc9
products9['Price after discount'] = price9
products9

Show Brand                        Description Price after discount
0              Nike       Men KD13 EP Basketball Shoes                12995
1              Nike         Men React Infinity Running                11621
2   PUMA Motorsport      Unisex Mercedes Running Shoes                 7999
3              Nike     Women PEGASUS 37 Running Shoes                 7496
4              Nike        Men JORDAN DELTA Basketball                12495
..              ...                                ...                  ...
95            Ruosh  Men Solid Leather Formal Slip-Ons                 6990
96             Geox         Men Leather Formal Oxfords                 7867
97     UNDER ARMOUR          Charged Pursuit 2 Running                 6999
98             Puma      Men Ignite Sock Plus Sneakers                 6749
99             Nike          Men REACT MILER 2 Running                 9770

[100 rows x 3 columns]

# Q10: 
Go to webpage https://www.amazon.in/

Enter “Laptop” in the search field and then click the search icon.
Then set CPU Type filter to “Intel Core i7” and “Intel Core i9”

After setting the filters scrape first 10 laptops data. You have to scrape 3 attributes
for each laptop:
1. title
2. Ratings
3. Price

In [31]:
driver10 = webdriver.Chrome(r"D:\Downloads\IDM Downloads\chromedriver.exe")
url10 = "https://www.amazon.in/"
driver10.get(url10)

In [32]:
# Searching for laptop in the webpage
search_item10 = driver10.find_element_by_id("twotabsearchtextbox")
search_item10.send_keys('Laptop')

# clicking the search button
search_button10 = driver10.find_element_by_id("nav-search-submit-button")
search_button10.click()

In [33]:
# Applying the filter for i7 processor
i7_filer10 = driver10.find_elements_by_xpath("//span[@class = 'a-size-base a-color-base']")
for i in i7_filer10:
    if i.text == 'Intel Core i7':
        i.click()
        break

In [34]:
# Scrapping for the laptop names
i7_lap_title10 = driver10.find_elements_by_xpath("//span[@class = 'a-size-medium a-color-base a-text-normal']")
i7_titles10 = []
for i in i7_lap_title10:
    i7_titles10.append(i.text)
i7_titles10 = i7_titles10[:10]
#i7_titles10

# Finding the urls for each laptop of i7 processor
i7_urls10 = []
i7_lap_urls10 = driver10.find_elements_by_xpath("//a[@class = 'a-link-normal a-text-normal']")
for i in i7_lap_urls10:
    i7_urls10.append(i.get_attribute("href"))
i7_urls10 = i7_urls10[:10]
#i7_urls10

In [35]:
from selenium.common.exceptions import NoSuchElementException # To handle no such element exception

# opening every url and scrapping for price and rating of laptop.
i7_price10 = []
i7_ratings10 = []

for i in i7_urls10:
    driver10.get(i)
    time.sleep(3)
    try: # for laptop price
        i7_lap_price10 = driver10.find_element_by_xpath("//td[@class = 'a-span12']")
        i7_price10.append(i7_lap_price10.text)
    except NoSuchElementException:
        i7_price10.append('-')
    
    try: # for laptop rating
        i7_lap_rating10 = driver10.find_element_by_xpath("//span[@data-hook = 'rating-out-of-text']")
        i7_ratings10.append(i7_lap_rating10.text.replace(' out of 5', ''))
    except NoSuchElementException:
        i7_ratings10.append('-')

        
#i7_price10
#i7_ratings10

In [36]:
i7_laptops10 = pd.DataFrame()
i7_laptops10['Laptop Name'] = i7_titles10
i7_laptops10['Laptop Price'] = i7_price10
i7_laptops10['Laptop Rating'] = i7_ratings10
i7_laptops10['Laptop Processor'] = 'i7'
i7_laptops10

Laptop Name   Laptop Price  \
0  Mi Notebook Horizon Edition 14 Intel Core i5-1...    ₹ 53,999.00   
1  Lenovo Legion 5Pi 10th Gen Intel Core i7 15.6"...  ₹ 1,35,490.00   
2  Life Digital Laptop 15.6-inch (39.62 cms) (Int...    ₹ 47,050.00   
3  HP Pavilion (2021) Thin & Light 11th Gen Core ...    ₹ 84,999.00   
4  Lenovo IdeaPad Flex 5 11th Gen Intel Core i7 1...    ₹ 83,990.00   
5  Asus ROG Zephyrus S Ultra Slim Gaming Laptop, ...  ₹ 3,43,099.00   
6  (Renewed) Dell Intel 4th Gen Core i7 4600U 14-...    ₹ 39,999.00   
7  CUK VivoBook K571 by ASUS 15 Inch Gaming Lapto...  ₹ 1,44,643.00   
8  ASUS VivoBook S S14 Intel Core i7-1165G7 11th ...    ₹ 81,990.00   
9  Lenovo Legion Y540 Intel Core i7 9th Gen 15.6”...    ₹ 68,990.00   

  Laptop Rating Laptop Processor  
0           4.4               i7  
1           4.3               i7  
2           2.9               i7  
3           4.4               i7  
4             4               i7  
5           4.2               i7  
6             -               i7  
7           4.7               i7  
8           4.4               i7  
9             4               i7

Similarly, for i9 processors, we scrap the data

In [40]:
driver10.get("https://www.amazon.in/s?k=Laptop&ref=nb_sb_noss_2")
time.sleep(3)

# Selecting filter for 19 processor
i9_filer10 = driver10.find_elements_by_xpath("//span[@class = 'a-size-base a-color-base']")
for i in i9_filer10:
    if i.text == 'Intel Core i9':
        i.click()
        break

In [41]:
# Scrapping for laptop names
i9_lap_title10 = driver10.find_elements_by_xpath("//span[@class = 'a-size-medium a-color-base a-text-normal']")
i9_titles10 = []
for i in i9_lap_title10:
    i9_titles10.append(i.text)
i9_titles10 = i9_titles10[:10]
#i9_titles10

# Finding the urls for each 19 processor laptop
i9_urls10 = []
i9_lap_urls10 = driver10.find_elements_by_xpath("//a[@class = 'a-link-normal a-text-normal']")
for i in i9_lap_urls10:
    i9_urls10.append(i.get_attribute("href"))
i9_urls10 = i9_urls10[:10]
#i9_urls10

In [42]:
i9_price10 = []
i9_ratings10 = []

# opening every url and scrapping for price and rating of laptop.
for i in i9_urls10:
    driver10.get(i)
    time.sleep(3)
    try: # for laptop price
        i9_lap_price10 = driver10.find_element_by_xpath("//td[@class = 'a-span12']")
        i9_price10.append(i9_lap_price10.text)
    except NoSuchElementException:
        i9_price10.append('-')
        
    try: # for laptop rating
        i9_lap_rating10 = driver10.find_element_by_xpath("//span[@data-hook = 'rating-out-of-text']")
        i9_ratings10.append(i9_lap_rating10.text.replace(' out of 5', ''))
    except NoSuchElementException:
        i9_ratings10.append('-')
        
#i9_price10
#i9_ratings10

In [43]:
i9_laptops10 = pd.DataFrame()
i9_laptops10['Laptop Name'] = i9_titles10
i9_laptops10['Laptop Price'] = i9_price10
i9_laptops10['Laptop Rating'] = i9_ratings10
i9_laptops10['Laptop Processor'] = 'i9'
i9_laptops10

Laptop Name   Laptop Price  \
0  ASUS ROG Zephyrus Duo 15, 15.6" FHD 300Hz/3ms,...  ₹ 2,66,990.00   
1  ASUS ROG G703GI-E5148T 17.3" (43.94 cms) FHD 1...  ₹ 4,13,890.00   
2  ASUS ZenBook Pro Duo Intel Core i9-10980HK 10t...  ₹ 2,99,325.00   
3  Dell XPS 9570 15.6" (39.62cms) UHD Laptop (8th...  ₹ 2,48,790.00   
4  Lenovo Legion 7 10th Gen Intel Core i9 15.6 in...  ₹ 2,62,990.00   
5  ASUS ZenBook Pro Duo UX581GV-H2041T Intel Core...  ₹ 2,79,990.00   
6  ASUS ROG Strix Scar 15 (2020), 15.6" FHD 300Hz...  ₹ 2,15,990.00   
7  ASUS ROG Strix Scar 17 (2020), 17.3" FHD 300Hz...  ₹ 2,77,639.00   
8  (Renewed) Dell XPS 9570 Laptop|i9-8950HK|32GB ...  ₹ 1,89,900.00   
9  Apple MacBook Pro (16-inch, 16GB RAM, 1TB Stor...  ₹ 2,15,990.00   

  Laptop Rating Laptop Processor  
0             -               i9  
1           3.7               i9  
2           3.8               i9  
3           2.4               i9  
4           3.7               i9  
5           3.9               i9  
6           4.2               i9  
7           4.8               i9  
8             -               i9  
9           4.2               i9

In [44]:
laptops10 = pd.concat([i7_laptops10, i9_laptops10], ignore_index = True)
laptops10

Laptop Name   Laptop Price  \
0   Mi Notebook Horizon Edition 14 Intel Core i5-1...    ₹ 53,999.00   
1   Lenovo Legion 5Pi 10th Gen Intel Core i7 15.6"...  ₹ 1,35,490.00   
2   Life Digital Laptop 15.6-inch (39.62 cms) (Int...    ₹ 47,050.00   
3   HP Pavilion (2021) Thin & Light 11th Gen Core ...    ₹ 84,999.00   
4   Lenovo IdeaPad Flex 5 11th Gen Intel Core i7 1...    ₹ 83,990.00   
5   Asus ROG Zephyrus S Ultra Slim Gaming Laptop, ...  ₹ 3,43,099.00   
6   (Renewed) Dell Intel 4th Gen Core i7 4600U 14-...    ₹ 39,999.00   
7   CUK VivoBook K571 by ASUS 15 Inch Gaming Lapto...  ₹ 1,44,643.00   
8   ASUS VivoBook S S14 Intel Core i7-1165G7 11th ...    ₹ 81,990.00   
9   Lenovo Legion Y540 Intel Core i7 9th Gen 15.6”...    ₹ 68,990.00   
10  ASUS ROG Zephyrus Duo 15, 15.6" FHD 300Hz/3ms,...  ₹ 2,66,990.00   
11  ASUS ROG G703GI-E5148T 17.3" (43.94 cms) FHD 1...  ₹ 4,13,890.00   
12  ASUS ZenBook Pro Duo Intel Core i9-10980HK 10t...  ₹ 2,99,325.00   
13  Dell XPS 9570 15.6" (39.62cms) UHD Laptop (8th...  ₹ 2,48,790.00   
14  Lenovo Legion 7 10th Gen Intel Core i9 15.6 in...  ₹ 2,62,990.00   
15  ASUS ZenBook Pro Duo UX581GV-H2041T Intel Core...  ₹ 2,79,990.00   
16  ASUS ROG Strix Scar 15 (2020), 15.6" FHD 300Hz...  ₹ 2,15,990.00   
17  ASUS ROG Strix Scar 17 (2020), 17.3" FHD 300Hz...  ₹ 2,77,639.00   
18  (Renewed) Dell XPS 9570 Laptop|i9-8950HK|32GB ...  ₹ 1,89,900.00   
19  Apple MacBook Pro (16-inch, 16GB RAM, 1TB Stor...  ₹ 2,15,990.00   

   Laptop Rating Laptop Processor  
0            4.4               i7  
1            4.3               i7  
2            2.9               i7  
3            4.4               i7  
4              4               i7  
5            4.2               i7  
6              -               i7  
7            4.7               i7  
8            4.4               i7  
9              4               i7  
10             -               i9  
11           3.7               i9  
12           3.8               i9  
13           2.4               i9  
14           3.7               i9  
15           3.9               i9  
16           4.2               i9  
17           4.8               i9  
18             -               i9  
19           4.2               i9